frontend/components/MainPage/hooks/useHealthCheck.ts

```diff
@@ -0,0 +1,54 @@
+import { message } from 'antd';
+import { isEmpty } from 'lodash';
+import { useEffect } from 'react';
+
+import { Pages } from '@/enums/Pages';
+import { useElectronApi } from '@/hooks/useElectronApi';
+import { usePageState } from '@/hooks/usePageState';
+import { useServices } from '@/hooks/useServices';
+import { delayInSeconds } from '@/utils/delay';
+
+/**
+ * Health check hook
+ */
+export const useHealthCheck = () => {
+  const { deploymentDetails, selectedService } = useServices();
+  const { goto: gotoPage } = usePageState();
+  const { showNotification } = useElectronApi();
+
+  const serviceId = selectedService?.service_config_id;
+
+  useEffect(() => {
+    const checkHealth = async () => {
+      if (!serviceId) return true;
+      if (!deploymentDetails) return true;
+
+      try {
+        const healthcheck = deploymentDetails.healthcheck;
+        if (isEmpty(healthcheck) || !healthcheck.env_var_status?.needs_update) {
+          return true;
+        }
+
+        // Check if there are any errors in the env vars
+        const envVarsErrors = Object.values(
+          healthcheck.env_var_status?.env_vars || {},
+        );
+        if (envVarsErrors.length === 0) return true;
+
+        const firstError = envVarsErrors[0];
+        showNotification?.(firstError);
+        message.error(firstError);
```

Comment:
could we add some logic that notification showing/navigation has been done not to do it again, since we request deployment endpoint by interval, deploymentDetails object will be new each time, I expect this to go into loop

Reply:
but if you want to get back on main (e.g update creds later/check balance/whatever) you'll go through this again?